packages/ember-htmlbars/tests/integration/component_lifecycle_test.js

```diff
@@ -344,6 +302,51 @@ QUnit.test('changing a component\'s displayed properties inside didInsertElement
   });
 });
 
+QUnit.test('changing an attributeBinding that is not provided in this.attrs triggers willRender', function() {
+  var service;
+
+  registry.register('service:foo-bar', Service.extend({
+    init() {
+      this._super(...arguments);
+      this.value = null;
+      service = this;
+    }
+  }));
+
+  registry.register('component:the-top', component('top', {
+    fooBar: inject.service('foo-bar'),
+
+    attributeBindings: ['dataFooBar:data-foo-bar'],
+
+    dataFooBar: computed('fooBar.value', function() {
+      return this.get('fooBar.value');
+    })
+  }));
+
+  registry.register('template:components/the-top', compile('Top'));
+
+  view = EmberView.extend({
+    template: compile('{{the-top}}'),
+    container: container
+  }).create();
+
```

Comment:
This test is testing components, so why is `scope.component` not set? Surely it goes through the `ComponentNodeManager`? Am I just misunderstanding the test?